tests/integration/base_deploy_test.go

```diff
@@ -140,6 +140,13 @@ func (op *TestCluster) SetInstallData(version string) {}
 
 // TearDownRook is a wrapper for tearDown after Suite
 func (op *TestCluster) Teardown() {
-	op.installer.GatherAllRookLogs(op.namespace, installer.SystemNamespace(op.namespace), op.installer.T().Name())
+	if op.installer.T().Failed() {
```

Comment:
As a pattern for gathering the logs, how about the following? If we always call the GatherAllRookLogs() method, then inside the method we can check for the failure or logs settings. For example:
```
func (h *CephInstaller) GatherAllRookLogs(namespace, systemNamespace string, testName string) {
  if !h.T().Failed() && h.Env.Logs != "all" {
    return
  }
  ...
}
```

Reply:
This looks like a better approach. I've updated the code to use this pattern.